File: week1/q2.c
#include <stdio.h>

int findMax(int arr[], int n) {
    int max = arr[0];  
    for (int i = 1; i < n; i++) {
        if (arr[i] > max) {
            max = arr[i];  
        }
    }
    return max;
}

int main() {
    int arr[100], n;

    printf("Enter the number of elements: ");
    scanf("%d", &n);

    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int largest = findMax(arr, n);
    printf("The largest element in the array is: %d\n", largest);

    return 0;
}

File: week1/q2.py
class Solution:
    def largest(self, arr):
        """
        Returns the largest element in the array.
        """
        if not arr:
            raise ValueError("Array is empty")

        max_element = arr[0]
        for num in arr[1:]:
            if num > max_element:
                max_element = num
        return max_element

# Sample usage:
obj = Solution()
arr = [1, 8, 7, 56, 90]
res = obj.largest(arr)
print("Output:", res)